File: frontend/src/domain/paymentStatus.ts
export const PAYMENT_STATUSES = [
  { value: 'pending', label: '확인 대기' },
  { value: 'deposit_paid', label: '계약금 확인' },
  { value: 'balance_pending', label: '잔금 확인 필요' },
  { value: 'paid', label: '완납' },
  { value: 'unpaid', label: '미수' },
  { value: 'refunded', label: '환불' },
];

export const PARTNER_PAYMENT_STATUSES = [
  { value: 'unpaid', label: '미지급' },
  { value: 'ready', label: '지급 대기' },
  { value: 'paid', label: '지급 완료' },
  { value: 'hold', label: '보류' },
];

export const PAYMENT_CHECK_STATUSES = ['pending', 'balance_pending', 'unpaid'];
export const BALANCE_INCOMPLETE_STATUSES = ['pending', 'deposit_paid', 'balance_pending', 'unpaid'];

export function paymentStatusLabel(status) {
  return PAYMENT_STATUSES.find((item) => item.value === status)?.label || status || '-';
}

export function partnerPaymentStatusLabel(status) {
  return PARTNER_PAYMENT_STATUSES.find((item) => item.value === status)?.label || status || '-';
}

export function isPaymentCheckNeeded(status) {
  return PAYMENT_CHECK_STATUSES.includes(status);
}

export function isBalanceIncomplete(status) {
  return BALANCE_INCOMPLETE_STATUSES.includes(status);
}

File: backend/app/domain/payment_status.py
from enum import StrEnum


class PaymentStatus(StrEnum):
    PENDING = "pending"
    DEPOSIT_PAID = "deposit_paid"
    BALANCE_PENDING = "balance_pending"
    PAID = "paid"
    UNPAID = "unpaid"
    REFUNDED = "refunded"


class PartnerPaymentStatus(StrEnum):
    UNPAID = "unpaid"
    READY = "ready"
    PAID = "paid"
    HOLD = "hold"


PAYMENT_CHECK_STATUSES: tuple[str, ...] = (
    PaymentStatus.PENDING,
    PaymentStatus.BALANCE_PENDING,
    PaymentStatus.UNPAID,
)

PAYMENT_DONE_STATUSES: tuple[str, ...] = (
    PaymentStatus.DEPOSIT_PAID,
    PaymentStatus.PAID,
)

PARTNER_SETTLEMENT_PENDING_STATUSES: tuple[str, ...] = (
    PartnerPaymentStatus.UNPAID,
    PartnerPaymentStatus.READY,
)

PAYMENT_TRACKED_FIELDS: tuple[str, ...] = (
    "total_amount",
    "discount_amount",
    "deposit_amount",
    "balance_amount",
    "onsite_extra_amount",
    "payment_status",
    "payment_memo",
    "evidence_memo",
    "receipt_type",
    "receipt_status",
    "partner_payment_amount",
    "partner_payment_status",
    "partner_settled_at",
    "broker_payment_amount",
    "broker_payment_status",
    "broker_settled_at",
    "customer_visible_payment",
)


def is_payment_check_needed(payment_status: str | None) -> bool:
    return payment_status in PAYMENT_CHECK_STATUSES
